docs/dependency-injection-liftweb-scala.adoc

```diff
@@ -0,0 +1,382 @@
+:idprefix:
+:idseparator: -
+:toc: right
+:toclevels: 2
+
+= Dependency Injection in Lift
+
+== Introduction
+
+Dependency injection is an important topic in the JVM world. With Java, we usually go with Guice
+or Spring whenever we need it. However, scala has certain advantages over Java, like functions,
+that allow basic DI without resorting to entire DI frameworks.
+
+Scala allows the use of http://jonasboner.com/real-world-scala-dependency-injection-di/[cake pattern]
+that can provide DI by composing traits.
+
+Although the cake pattern is helpful, it doesn't provide complete DI functionality.
```

Comment:
This phrasing is a bit awkward. In general, single-sentence paragraphs get wobbly. Perhaps something like:

> Although [the cake pattern] can be used to provide DI by composing traits, it doesn't provide complete DI functionality. In particular, though this pattern allows you to compose complex classes out of Scala traits, it's less helpful in terms of all allowing you to make dynamic choices about which combination of dependencies to vend in a given situation.